src/Kafker/Commands/EmitCommand.cs

```diff
@@ -16,55 +13,29 @@ public class EmitCommand : IEmitCommand
         private readonly IConsole _console;
         private readonly KafkerSettings _settings;
         private readonly IProducerFactory _producerFactory;
+        private readonly IEventsEmitter _eventsEmitter;
 
-        public EmitCommand(IConsole console, KafkerSettings settings, IProducerFactory producerFactory)
+        public EmitCommand(IConsole console, KafkerSettings settings, IProducerFactory producerFactory, IEventsEmitter eventsEmitter)
         {
             _console = console;
             _settings = settings;
             _producerFactory = producerFactory;
+            _eventsEmitter = eventsEmitter;
         }
 
-        public async Task<int> InvokeAsync(CancellationToken cancellationToken, string topic, string fileName)
+        public async Task<int> InvokeAsync(CancellationToken cancellationToken, string topic, string fileName )
         {
-            var cfg = await ExtractorHelper.ReadConfigurationAsync(topic, _settings, _console);
-
-            var producedEvents = 0;
-            if (!File.Exists(fileName))
-            {
-                fileName = Path.Combine(_settings.Destination, fileName);
-                if (!File.Exists(fileName))
-                {
-                    await _console.Error.WriteAsync($"Error: Cannot find the file: {fileName}");
-                    return await Task.FromResult(1).ConfigureAwait(false); // no ok
-                }
-            }
-
-            using var topicProducer = _producerFactory.Create(cfg);
-            using var reader = new StreamReader(fileName);
             try
             {
-                string line;
-                while ((line = await reader.ReadLineAsync()) != null)
-                {
-                    if (cancellationToken.IsCancellationRequested) break;
-
-                    var pair = line.Split("|");
-                    var jsonText = pair[1].Substring(1, pair[1].Length - 2);
-                    await topicProducer.ProduceAsync(jsonText);
-                    producedEvents++;
-                }
-            }
-            catch (Exception err)
-            {
-                await _console.Error.WriteLineAsync($"\r\nError: {err.Message}");
-                return await Task.FromResult(1).ConfigureAwait(false); // error
+                await _eventsEmitter.EmitEvents(cancellationToken,fileName,topic);
+               
             }
-            finally
+            catch (Exception e)
             {
-                await _console.Out.WriteLineAsync($"\r\nProduced {producedEvents} events");
+                await _console.Out.WriteLineAsync(e.Message);
```

Comment:
Use exception handling which was before the change.
It should wrap the line and use 1 as a result